internal/connectors/engine/workflow/fetch_accounts.go

```diff
@@ -162,6 +162,18 @@ func (w Workflow) fetchAccounts(
 		}
 
 		hasMore = accountsResponse.HasMore
+
+		if w.shouldContinueAsNew(ctx) {
+			// If we have lots and lots of accounts, sometimes, we need to
+			// continue as new to not exeed the maximum history size or length
+			// of a workflow.
+			return workflow.NewContinueAsNewError(
```

Comment:
Since this returns an error, I guess that means that any data we fetched in this run, will have to be refetched from the PSP, right? Could we maybe check if a new workflow is recommended _before_ making any calls to the PSP?

Reply:
No it's not the case, since the state was inserted just before, we will start where we stopped !

Reply:
Ah ok, yes you're right 👍